app/src/androidTest/java/com/github/se/project/ui/lesson/ConfirmedLessonTest.kt

```diff
@@ -0,0 +1,179 @@
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mockito.*
+import org.mockito.kotlin.whenever
+
+@RunWith(AndroidJUnit4::class)
+public class ConfirmedLessonTest {
+
+  @get:Rule val composeTestRule = createComposeRule()
+  @get:Rule
+  val permissionRule: GrantPermissionRule =
+      GrantPermissionRule.grant(android.Manifest.permission.ACCESS_FINE_LOCATION)
+
+  val mockLessonRepository = mock(LessonRepository::class.java)
+
+  // Mock dependencies
+  val mockProfilesRepository = mock(ProfilesRepository::class.java)
+  val mockNavigationActions = mock(NavigationActions::class.java)
+  val lessonViewModel = LessonViewModel(mockLessonRepository)
+  val listProfilesViewModel = ListProfilesViewModel(mockProfilesRepository)
+
+  private val tutorProfile =
+      Profile(
+          uid = "12345",
+          googleUid = "67890",
+          firstName = "John",
+          lastName = "Doe",
+          phoneNumber = "1234567890",
+          role = Role.TUTOR,
+          section = Section.GM,
+          academicLevel = AcademicLevel.MA2,
+          schedule = List(7) { List(12) { 0 } },
+          price = 50)
+
+  private val studentProfile =
+      Profile(
+          uid = "1",
+          googleUid = "67890",
+          firstName = "James",
+          lastName = "Donovan",
+          phoneNumber = "1234567890",
+          role = Role.STUDENT,
+          section = Section.GM,
+          academicLevel = AcademicLevel.BA2,
+          schedule = List(7) { List(12) { 0 } },
+          price = 50)
+
+  private val lesson =
+      Lesson(
+          id = "lesson1",
+          title = "Math Lesson",
+          timeSlot = "30/12/2024T14:00:00",
+          tutorUid = listOf("tutor1"),
+          studentUid = "1",
+          latitude = 37.7749,
+          longitude = -122.4194,
+          status = LessonStatus.CONFIRMED)
+
+  @Before
+  fun setUp() {
+    whenever(mockProfilesRepository.getProfiles(org.mockito.kotlin.any(), org.mockito.kotlin.any()))
+        .thenAnswer { invocation ->
+          val onSuccess = invocation.arguments[0] as (List<Profile>) -> Unit
+          onSuccess(
+              listOf(
+                  tutorProfile,
+                  studentProfile)) // Simulate a list of profiles with our beloved Ozymandias
+        }
+
+    whenever(
+            mockLessonRepository.getLessonsForTutor(
+                org.mockito.kotlin.eq(tutorProfile.uid),
+                org.mockito.kotlin.any(),
+                org.mockito.kotlin.any()))
+        .thenAnswer { invocation ->
+          val onSuccess = invocation.arguments[1] as (List<Lesson>) -> Unit
+          onSuccess(listOf(lesson))
+        }
+
+    listProfilesViewModel.getProfiles()
+    lessonViewModel.getLessonsForTutor(tutorProfile.uid, {})
+
+    lessonViewModel.selectLesson(lesson)
+    listProfilesViewModel.setCurrentProfile(tutorProfile)
+    Log.e("LeProfileBG", "${listProfilesViewModel.currentProfile}")
```

Comment:
Maybe remove this log for next time

Reply:
You're right but for CI reasons I want to fix this next time or tell to a teammate to remove it in his PR, as by mircale I finally pass the CI.

Reply:
Will be fixed soon in Léa's PR.